src/admin/components/modal/AddUserModalComponent.tsx

```diff
@@ -56,87 +62,182 @@ class AddUserModalComponent extends React.Component <any, any> {
 		return options;
 	}
 
-	componentWillReceiveProps(nextProps){}
+	makeSelectDom(optionsArr) {
+		return optionsArr.map((optionItem) =>
+			<option
+				key={optionItem} value={optionItem}>{ capitalizeFirstLetter(optionItem) }
+			</option>);
+	}
 
-	updateInputValue(event) {
-		const options = this.getCurrentInput(event.target.name);
-		options.input[event.target.name].value = event.target.value;
-		UIActions.modalWindowsChangeState(this.props.windowName, options);
+	getUserRoles() {
+		const roles = [
+			UserRole.SystemAdmin,
+			UserRole.SiteAdmin,
+			UserRole.Editor,
+			UserRole.Normal,
+		];
+		return this.makeSelectDom(roles);
+	}
+
+	hasNameError(): string | boolean {
+		if (this.props.name.value.length < 4) {
+			return 'Name has to be more than 3 symbols';
+		}
+		return false;
+	}
+
+	hasEmailError(): string | boolean {
+		if (this.props.email.value.length < 4) {
+			return 'Please enter valid email address';
+		}
+		return false;
+	}
+
+	hasPasswordError(): string | boolean {
+		if (this.props.userId) {
+			return false;
+		}
+		if (this.props.password.value.length < 4) {
+			return 'Password has to be more than 3 symbols';
+		}
+		return false;
+	}
+
+	isFormValid(): boolean {
+		return (!this.hasNameError() && !this.hasEmailError() && !this.hasPasswordError());
 	}
-	updateSelectValue(event){
+
+	updateInputValue(event): void {
 		const options = this.getCurrentInput(event.target.name);
-		options.input[event.target.name].value = +event.target.value;
+		options.input[event.target.name].value = event.target.value;
 		UIActions.modalWindowsChangeState(this.props.windowName, options);
 	}
 
-	closePopup() : void {
+	closePopup(): void {
 		if (this.props.userId){
 			UIActions.closeReset(this.props.windowName);
 		} else {
 			UIActions.modalWindowsChangeState(this.props.windowName, {isOpen: false});
 		}
 	}
-
-	saveUser(event): void {}
-
-	closeReset() : void {
-		UIActions.closeReset(this.props.windowName);
+	onFocus(event) {
+		const name = event.target.name;
+		this.props[name].touched = true;
+	}
+	onBlur(event) {
+		const options = this.getCurrentInput(event.target.name);
+		UIActions.modalWindowsChangeState(this.props.windowName, options);
+	}
+	saveUser(event): void {
+		event.preventDefault();
```

Comment:
const {
           email: {value: email},
           password: {value: password},
           userId: id,
           name: {value: name},
           role: {value: role}
} = this.props;
const data = {email, password, id, name, role};